### DISCUSSION TITLE
List content type - cannot update SchemaXml property

### DISCUSSION THREAD

**Discussion by tomasz-sintel** [pnp/pnpcore#1418] [Q&A]
I am trying to update a list content type. Primarily I wanted to do this because of missing properties in the ContentType model, but I can't even get it to work with the simplest property. Please help me out.
My code:

``` csharp
var list = await context.Web.Lists.GetByIdAsync(request.ListId, l => l.ContentTypes);
await list.LoadAsync(item => item.ContentTypes.QueryProperties(ct => ct.All));
var contentTypes = list.ContentTypes;

foreach (var contentType in contentTypes.AsRequested().Where(ct => ct.Name == "Item"))
{
    //contentType.Description = $"Create a new list item {DateTime.Now:HH:mm:ss}";
    contentType.SchemaXml = GetUpdatedContentTypeSchemaXml(contentType.SchemaXml, true);
    await contentType.UpdateAsync();
}

await list.UpdateAsync();
```
At the moment, the GetUpdatedContentTypeSchemaXml only modifies the Description attribute in the SchemaXml, nothing more. I am comparing the property value before and after the setter.

It does not matter if I update the list at the end or not.

I do have sufficient permissions on the list, I can update the CT through normal properties (see commented line).

Am I missing something?

**Comment by jansenbe** [comment]
@tomasz-sintel : I just cross checked the SharePoint code base and the ContentType.SchemaXml property is not settable, so it's probably just ignored. I'll update PnP Core SDK to reflect that. I did notice that the ContentType.SchemaXmlWithResourceTokens property is settable.
